### ISSUE TITLE
The members data and errors MUST NOT coexist in the same document.

### ISSUE THREAD

**Issue by Unisay** [zalando/scala-jsonapi#45]
At the moment, `RootObject` is

```scala
  case class RootObject(
    data: Option[RootObject.Data] = None,
    links: Option[Links] = None,
    errors: Option[Errors] = None,
    meta: Option[Meta] = None,
    included: Option[Included] = None,
    jsonApi: Option[JsonApi] = None)
```

However, JSON-API spec clearly says:
http://jsonapi.org/format/#document-top-level

> The members data and errors MUST NOT coexist in the same document.

Suggestion:

```scala
  case class RootObject(
    body: Either[Errors, RootObject.Data], // <<<< 
    links: Option[Links] = None,
    meta: Option[Meta] = None,
    included: Option[Included] = None,
    jsonApi: Option[JsonApi] = None)
```

**Comment by asgoncalves** [comment]
we are aware of this problem.
see issue #6

**Comment by LappleApple** [comment]
Hey @asgoncalves, can we close this issue? It dates back to May 2016.

**Comment by oporkka** [comment]
In my opinion this should stay open since the code has not been yet fixed and this is a sub-issue of #6 which is way too generic. I'd divide it to sub-issues like this or otherwise it'll become a dinosaur that we'll never fix.

**Comment by LappleApple** [comment]
@oporkka Yes, splitting things into smaller to-do's is a great idea. How would you like to go about it?

**Comment by oporkka** [comment]
My proposal is to keep #6 open as an epic and gather the smaller issues under it. We can edit the description there and list the sub-issues there. It would be great if someone who knows could list what parts of the spec are actually missing. However we can also list the smaller issues there as they come up based on people's needs and observations.
